File: server/prisma/seed/dataUsage.ts
import { DataUsages, PrismaClient } from '@prisma/client';

const testDataUsages: DataUsages[] = [
  {
    dataUsageId: 1,
    routerId: 1,
    simId: 1,
    userId: 1,
    dataUsage: 425336n,
    dateId: 20231115,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
  {
    dataUsageId: 2,
    routerId: 2,
    simId: 2,
    userId: 1,
    dataUsage: 687336n,
    dateId: 20251115,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
  {
    dataUsageId: 3,
    routerId: 3,
    simId: 3,
    userId: 1,
    dataUsage: 117342n,
    dateId: 20231115,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
  {
    dataUsageId: 4,
    routerId: 4,
    simId: 4,
    userId: 1,
    dataUsage: 452763n,
    dateId: 20231115,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
  {
    dataUsageId: 5,
    routerId: 5,
    simId: 5,
    userId: 1,
    dataUsage: 247865n,
    dateId: 20231115,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
  {
    dataUsageId: 6,
    routerId: 6,
    simId: 6,
    userId: 1,
    dataUsage: 147836n,
    dateId: 20231116,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  },
];

export async function seedDataUsages(prisma: PrismaClient) {
  const dataUsages = await prisma.dataUsages.createMany({
    data: testDataUsages,
    skipDuplicates: true,
  });
}

File: client/lib/models/data_usage.dart
class DataUsage {
  final int dataUsageId;
  final int dateId;
  final int? routerId;
  final int simId;
  final int userId;
  final String dataUsage;
  final String? createdAt;
  final String? updatedAt;
  final String? deletedAt;

  DataUsage({
    required this.dataUsageId,
    required this.dateId,
    this.routerId,
    required this.simId,
    required this.userId,
    required this.dataUsage,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
  });

  factory DataUsage.fromJson(Map<String, dynamic> json) {
    return DataUsage(
      dataUsageId: json['data_usage_id'],
      dateId: json['date_id'],
      routerId: json['router_id'],
      simId: json['sim_id'],
      userId: json['user_id'],
      dataUsage: json['data_usage'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
      deletedAt: json['deleted_at'],
    );
  }
}
